### PR TITLE
Labeling

### LINKED ISSUE #28: Create Text Object at the end of label-editing

### PR DESCRIPTION
**Pull request by tim-kilian**
Closes #28 

## Describe your changes
- Reworked meta-label-editing
- CreateModule now creates all type of elements
- DoubleClick to edit label

## Screenshots
labeling

### CODE REVIEW

File: src/features/create/tools/CreateTool.js [hunk 1]
```diff
@@ -9,42 +10,72 @@ export class CreateTool extends Tool {
 
     onDisable (event) {
         this.eventBus.fire('factory.reset', event);
-        this.eventBus.fire('create.cursor.unset', event);
-        this.eventBus.fire('create.preview.clear', event);
-        this.eventBus.fire('create.marker.clear', event);
```
Comment (0x4d) [inline src/features/create/tools/CreateTool.js:14]: I like this change.

File: src/features/edit/providers/EditProvider.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+                height: element.height,
+            },
+            style: {
+                border: '2px dashed #ccc',
+            },
+            text: element.text,
+            options: {
+                resizable: true,
+                centerVertically: true,
+                autoActivate: true,
```
Comment (0x4d) [inline src/features/edit/providers/EditProvider.js:38]: Aah! :smile:

File: src/features/edit/tools/EditTool.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    onMouseMove (event) {
+    }
+
+    onMouseUp (event) {
+        const editingContent = this.directEditing._textbox.parent.firstChild;
+        if (!editingContent.innerHTML) {
+            editingContent.innerHTML = '<br>';
+        }
+        editingContent.focus();
+        editingContent.focus();
```
Comment (0x4d) [inline src/features/edit/tools/EditTool.js:41]: Duplicate line?

**Review by 0x4d — APPROVED** [review]
Very sensible changes. Great to finally have the ability to create labels.

File: src/features/edit/providers/EditProvider.js (hunk 2 above)
Comment (tim-kilian) [reply src/features/edit/providers/EditProvider.js]: True, I had some problems typing after focus and tried multiple things. But it seems to work, this one could be some leftover.

File: src/features/edit/tools/EditTool.js (hunk 3 above)
Comment (tim-kilian) [reply src/features/edit/tools/EditTool.js:41]: True, I had some problems typing after focus and tried multiple things. But it seems to work, this one could be some leftover.

**Pull request merged**